.devcontainer/devcontainer.json

```diff
@@ -1,15 +1,32 @@
 {
 	"name": "leeway",
-	"image": "mcr.microsoft.com/devcontainers/base:ubuntu",
+	"build": {
+		"dockerfile": "Dockerfile"
+	},
 	"features": {
-		"ghcr.io/devcontainers/features/node:1": {},
-		"ghcr.io/devcontainers/features/go:1": {},
-		"ghcr.io/devcontainers/features/common-utils:2": {},
-		"ghcr.io/devcontainers-contrib/features/shfmt:1": {
-			"version": "3.10.0"
+		"ghcr.io/devcontainers/features/node:1": {
+			"version": "18"
+		},
+		"ghcr.io/devcontainers/features/go:1": {
+			"version": "1.24"
 		},
+		"ghcr.io/devcontainers/features/common-utils:2": {},
 		"ghcr.io/dhoeric/features/google-cloud-cli:1": {},
-		"ghcr.io/devcontainers/features/aws-cli:1": {},
+		"ghcr.io/devcontainers/features/aws-cli:1": {}
 	},
-	"postStartCommand": "./.devcontainer/install-leeway.sh"
```

Comment:
we ask the maintainer install global tools in Dockerfile for caching, I do wonder whether it applies here